### DISCUSSION TITLE
Why does FieldInput allow a Slot? - causing SSR issue

### DISCUSSION THREAD

**Discussion by RomainLanz** [chakra-ui/ark#3158] [Q&A] [answered]
### Description

Hey there! 👋🏻 

The `FieldInput` component currently allows a slot, which is problematic because the `<input>` element is a void element, meaning it cannot have child elements or content according to the HTML specification ([MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#technical_summary)).

Allowing a slot in the `FieldInput` component creates issues, especially during SSR (Server-Side Rendering), as the slot content is removed on the client side, leading to mismatches between the server-rendered HTML and the client-rendered HTML.

**I am happy to submit a pull request** if you allow me to remove the slot.

### Link to Reproduction (or Detailed Explanation)

N/A

### Steps to Reproduce

N/A

### Ark UI Version

4.5

### Framework

- [ ] React
- [ ] Solid
- [X] Vue

**Comment by RomainLanz** [comment]
It seems that a similar issue exists with the `textarea` component, where a slot is being used inside the element. Like the `<input>` element, the `<textarea>` element should not contain a slot because its content is meant to be the text between the opening and closing tags, rather than child elements.

Allowing a slot inside a `<textarea>` also creates an SSR mismatch, as the content inside the slot gets removed on the client during hydration, causing inconsistencies between the server-rendered and client-rendered HTML.

You can see in the compiled code below that the slot is passed as a string:

```ts
return (_ctx, _cache) => {
  return openBlock(), createElementBlock("textarea", mergeProps({ ref: unref(autoresizeTextarea) }, unref(field).getTextareaProps(), {
    value: _ctx.modelValue,
    onInput: _cache[0] || (_cache[0] = (event) => emit("update:modelValue", event.target.value)),
    style: props.autoresize ? { resize: "none", overflow: "hidden" } : void 0
  }), "    <slot />\n  ", 16, _hoisted_1);
};
```

**Comment by segunadebayo** [comment]
It allows a slot because of composition via the `asChild` prop. You can read more about it here:
https://ark-ui.com/react/docs/guides/composition

```tsx
<Field.Input asChild>
   <input />
</Field.Input>
```

Perhaps we should focus on resolving the SSR issue instead.

**Reply by RomainLanz** [reply]
> Perhaps we should focus on resolving the SSR issue instead.

Do you have any thoughts on how we might approach that?

**Reply by segunadebayo** [reply]
We should start with a reproduction first.

Perhaps, also using the `useFieldContext` directly might help you quickly get around it.

**Comment by segunadebayo** [comment, marked as answer]
This has been fixed in the Vue factory. Self-closing tags (`input`, `textarea`, etc.) no longer render slot content when `asChild` is false, preventing the SSR hydration mismatch:

```ts
// factory.ts
const SELF_CLOSING_TAGS = 'br, hr, img, input, area, textarea'.split(', ')

// When not using asChild, void elements pass undefined instead of slot content
if (!props.asChild) return () => h(component, attrs, isSelfClosingTag(component) ? undefined : slots.default?.())
```

The `<slot />` in the template is still there for `asChild` composition, but it's safely ignored for void elements during SSR.
